### DISCUSSION TITLE
Displaying Application Version on Cypress

### DISCUSSION THREAD

**Discussion by turalabdulsamad** [cypress-io/cypress#28412] [Questions and Help]
I would like to display the version number of my application on cypress. Is there any way to update the DOM? So, I can place the version number of the application on the header next to the cypress version?

![image](https://github.com/cypress-io/cypress/assets/49013404/a889a114-d0ea-48e0-aa27-7aa64c4a35b2)

**Comment by portyu9** [comment]
Hi @turalabdulsamad 

There is no supported Cypress API for modifying the Cypress Runner's own header DOM and adding arbitrary UI next to the Cypress version.

I would avoid reaching into Cypress's internal runner DOM because that UI is not a public API and can change between releases.

If the goal is simply to make the application version visible while debugging, add it to the Cypress Command Log instead:

```js
before(() => {
  Cypress.log({
    name: 'APP VERSION',
    message: '1.2.3',
  })
})
```

Or, if the application exposes its version through an endpoint:

```js
before(() => {
  cy.request('/version').then(({ body }) => {
    Cypress.log({
      name: 'APP VERSION',
      message: body.version,
    })
  })
})
```

Another good option is to expose the version in the **application's own DOM**:

```html
<meta name="app-version" content="1.2.3">
```

and assert/log it:

```js
cy.get('meta[name="app-version"]')
  .invoke('attr', 'content')
  .then((version) => {
    Cypress.log({
      name: 'APP VERSION',
      message: version,
    })
  })
```

That keeps the test independent of Cypress's private runner implementation.

Please mark this answer as accepted if it helped, thank you.
